north/src/runtime/mount.rs

```diff
@@ -431,3 +437,32 @@ async fn wait_for_file_deleted(path: &Path, timeout: time::Duration) -> Result<(
         .and_then(|r| r.map_err(Error::JoinError))
         .and_then(|r| r)
 }
+
+fn is_version_supported(manifest: &Manifest) -> bool {
+    const V1: Version = Version(semver::Version {
+        major: 1,
+        minor: 0,
+        patch: 0,
+        pre: vec![],
+        build: vec![],
+    });
+    const SUPPORTED_MIN_VERSION: &Version = &V1;
+    const KNOWN_MAX_VERSION: &Version = &V1;
+    const DEFAULT_VERSION: &Version = &V1; // used if manifest has no 'manifest_version' entry
+    let manifest_version = manifest
+        .manifest_version
+        .as_ref()
+        .unwrap_or(DEFAULT_VERSION);
+
+    // compare major versions
+    if manifest_version.0.major < SUPPORTED_MIN_VERSION.0.major {
+        return false;
+    }
+    if manifest_version.0.major > KNOWN_MAX_VERSION.0.major {
+        debug!(
+            "Warning: unknown manifest major version {}",
+            manifest_version
+        );
+    }
```

Comment:
1.case: npk is too old => not supported by newer runtime
2.case: runtime is too old => npk is not backward compatible (should not happen) but still in this case we should not continue

Reply:
Agreed. We should return false in the second case.